webapp/src/store/attachments.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSlice, PayloadAction} from '@reduxjs/toolkit'
+
+import {AttachmentBlock} from '../blocks/attachmentBlock'
+
+import {loadBoardData, initialReadOnlyLoad} from './initialLoad'
+
+import {RootState} from './index'
+
+type AttachmentsState = {
+    attachments: {[key: string]: AttachmentBlock}
+    attachmentsByCard: {[key: string]: AttachmentBlock[]}
+}
+
+const attachmentSlice = createSlice({
+    name: 'attachments',
+    initialState: {attachments: {}, attachmentsByCard: {}} as AttachmentsState,
+    reducers: {
+        updateAttachments: (state, action: PayloadAction<AttachmentBlock[]>) => {
+            for (const attachment of action.payload) {
+                if (attachment.deleteAt === 0) {
+                    state.attachments[attachment.id] = attachment
+                    if (!state.attachmentsByCard[attachment.parentId]) {
+                        state.attachmentsByCard[attachment.parentId] = [attachment]
+                        return
+                    }
```

Comment:
This is required when we are adding the attachment to the card for the first time. Else if we directly tried this  `state.attachmentsByCard[attachment.parentId].push(attachment)` it will throw an error.